gobblin-core/src/main/java/gobblin/source/extractor/filebased/FileBasedExtractor.java

```diff
@@ -174,12 +186,12 @@ public long getHighWatermark() {
    */
   @SuppressWarnings("unchecked")
   public Iterator<D> downloadFile(String file) throws IOException {
-    LOGGER.info("Beginning to download file: " + file);
+    LOG.info("Beginning to download file: " + file);
 
     try {
       InputStream inputStream = this.closer.register(this.fsHelper.getFileStream(file));
       Iterator<D> fileItr = (Iterator<D>) IOUtils.lineIterator(inputStream, ConfigurationKeys.DEFAULT_CHARSET_ENCODING);
-      if (workUnitState.getPropAsBoolean(ConfigurationKeys.SOURCE_SKIP_FIRST_RECORD, false) && fileItr.hasNext()) {
+      if (this.workUnitState.getPropAsBoolean(ConfigurationKeys.SOURCE_SKIP_FIRST_RECORD, false) && fileItr.hasNext()) {
```

Comment:
Please move the access to the config property to the constructor and initialize a member variable. Any idea why we event have a property like that? Seems very dangerous to me.